`k8s.io/kubectl/pkg/cmd/version` `TestNewCmdVersionWithoutConfigFile` always fails locally

/sig cli

run `make test` (or just `make test WHAT=k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/versionn`).

observe failure:

```
--- FAIL: TestNewCmdVersionWithoutConfigFile (0.06s)
    version_test.go:43: Cannot execute version command: could not parse pre-release/metadata (-master+$Format:%H$) in version "v0.0.0-master+$Format:%H$"
FAIL
FAIL	k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/version	0.307s
```

xref: https://github.com/kubernetes/kubernetes/issues/102607#issuecomment-854985308

--- a/staging/src/k8s.io/kubectl/pkg/cmd/version/version_test.go
+++ b/staging/src/k8s.io/kubectl/pkg/cmd/version/version_test.go
@@ -23,11 +23,12 @@ import (
 	"github.com/spf13/cobra"
 
 	"k8s.io/cli-runtime/pkg/genericclioptions"
-	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	cmdtesting "k8s.io/kubectl/pkg/cmd/testing"
 )
 
-func TestNewCmdVersionWithoutConfigFile(t *testing.T) {
-	tf := cmdutil.NewFactory(&genericclioptions.ConfigFlags{})
+func TestNewCmdVersionClientVersion(t *testing.T) {
+	tf := cmdtesting.NewTestFactory().WithNamespace("test")
+	defer tf.Cleanup()
 	streams, _, buf, _ := genericclioptions.NewTestIOStreams()
 	o := NewOptions(streams)
 	if err := o.Complete(tf, &cobra.Command{}); err != nil {
@@ -36,10 +37,7 @@ func TestNewCmdVersionWithoutConfigFile(t *testing.T) {
 	if err := o.Validate(); err != nil {
 		t.Errorf("Unexpected error: %v", err)
 	}
-	// FIXME soltysh:
-	// since we have defaulting to localhost:8080 in staging/src/k8s.io/client-go/tools/clientcmd/client_config.go#getDefaultServer
-	// we need to ignore the localhost:8080 server, when above gets removed this should be dropped too
-	if err := o.Run(); err != nil && !strings.Contains(err.Error(), "localhost:8080") {
+	if err := o.Run(); err != nil {
 		t.Errorf("Cannot execute version command: %v", err)
 	}
 	if !strings.Contains(buf.String(), "Client Version") {

